test/test_indexing.py

```diff
@@ -866,6 +880,18 @@ def test_bool_indices(self, device):
             )
             self.assertEqual(len(w), 1)
 
+    def test_list_indices(self, device):
```

Comment:
nit: we can use with self.assertWarns (or similar) to ensure we do through the warning?